aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -90,28 +91,35 @@ private static AwsClientFactory loadClientFactory(String impl, Map<String, Strin
 
   static class DefaultAwsClientFactory implements AwsClientFactory {
     private AwsProperties awsProperties;
+    private AwsClientProperties awsClientProperties;
+    private S3FileIOProperties s3FileIOProperties;
+    private HttpClientProperties httpClientProperties;
 
     DefaultAwsClientFactory() {
       awsProperties = new AwsProperties();
+      awsClientProperties = new AwsClientProperties();
+      s3FileIOProperties = new S3FileIOProperties();
+      httpClientProperties = new HttpClientProperties();
     }
 
     @Override
     public S3Client s3() {
-      return S3Client.builder()
-          .applyMutation(awsProperties::applyClientRegionConfiguration)
-          .applyMutation(awsProperties::applyHttpClientConfigurations)
-          .applyMutation(awsProperties::applyS3EndpointConfigurations)
-          .applyMutation(awsProperties::applyS3ServiceConfigurations)
-          .applyMutation(awsProperties::applyS3CredentialConfigurations)
-          .applyMutation(awsProperties::applyS3SignerConfiguration)
-          .build();
+      S3ClientBuilder builder = S3Client.builder();
+      builder.applyMutation(awsClientProperties::applyClientRegionConfiguration);
```

Comment:
Nit: I prefer just chaining them as before.

Reply:
+1 to that

Reply:
Sure, no problem, let me use the chaining, I'm updating the PR accordingly.